In [1]:
pip install pymongo

Note: you may need to restart the kernel to use updated packages.


In [2]:
import pymongo
from pymongo import MongoClient
import json
from pathlib import Path

def get_client():
    try:
        client = MongoClient(host = "mongodb://localhost:27017/")
    except:
        print ("connection error")
    return client

In [3]:
client = get_client()

db = client["gym"]

gym = db["Gymnases"]
sp = db["Sportifs"]

1.1) Afficher toutes les collections de la base

In [4]:

print(db.list_collection_names())

['Gymnases', 'Sportifs']


1.2) Compter le nombre de documents ds chacune des collections

In [5]:
print('Taille de la collection Gymnase : {}'.format(gym.find().count()))
print('Taille de la collection Sportifs : {}'.format(sp.find().count()))

Taille de la collection Gymnase : 28
Taille de la collection Sportifs : 150


1.2) Afficher un document de chacune des collections

2) On cherche le (ou les) conseiller(s) s’appelant "KERVADEC" ?

In [6]:
request = sp.find({
    "IdSportifConseiller" : 1, 
    "Nom" : "KERVADEC"
})

for element in request :
    print(element)

{'_id': '566eec5f662b388eba464204', 'IdSportif': 2, 'Nom': 'KERVADEC', 'Prenom': 'Yann', 'Sexe': 'M', 'Age': 28, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Basket ball', 'Volley ball', 'Ping pong', 'Football'], 'Arbitrer': ['Hockey', 'Football'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Tennis', 'Hockey', 'Badmington', 'Ping pong', 'Boxe']}}


3) On cherche les sportifs qui ont ou bien 32 ans ou 40 ans ?

In [7]:
request = sp.find({
    "$or" : [
        {"Age" : 32}, 
        {"Age" : 40}
    ]
})

for element in request :
    print(element)

{'_id': '566eec5f662b388eba464206', 'IdSportif': 4, 'Nom': 'DORLEANS', 'Prenom': 'Jean-michel', 'Sexe': 'M', 'Age': 32, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Volley ball', 'Football'], 'Arbitrer': ['Basket ball', 'Volley ball', 'Ping pong', 'Boxe'], 'Entrainer': ['Basket ball', 'Ping pong', 'Boxe']}}
{'_id': '566eec5f662b388eba46421d', 'IdSportif': 28, 'Nom': 'RABAHI', 'Prenom': 'Rabah', 'Sexe': 'M', 'Age': 40, 'IdSportifConseiller': 4, 'Sports': {'Jouer': ['Basket ball', 'Volley ball', 'Hand ball', 'Ping pong', 'Football']}}
{'_id': '566eec60662b388eba46423d', 'IdSportif': 60, 'Nom': 'TIZEGHAT', 'Prenom': 'Benamar', 'Sexe': 'M', 'Age': 32, 'IdSportifConseiller': 3, 'Sports': {'Jouer': ['Hand ball', 'Tennis', 'Football'], 'Arbitrer': 'Volley ball', 'Entrainer': ['Volley ball', 'Tennis', 'Ping pong']}}
{'_id': '566eec60662b388eba46423f', 'IdSportif': 62, 'Nom': 'BAZOUD', 'Prenom': 'Jérôme', 'Sexe': 'M', 'Age': 32, 'IdSportifConseiller': 3, 'Sports': {'Jouer': 'Football'}}
{'_i

4) On cherche les sportifs jouant au Basket-ball ?

In [8]:
request = sp.find({
    "Sports.Jouer" : "Basket ball"
})

for element in request :
    print(element)

{'_id': '566eec5f662b388eba464204', 'IdSportif': 2, 'Nom': 'KERVADEC', 'Prenom': 'Yann', 'Sexe': 'M', 'Age': 28, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Basket ball', 'Volley ball', 'Ping pong', 'Football'], 'Arbitrer': ['Hockey', 'Football'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Tennis', 'Hockey', 'Badmington', 'Ping pong', 'Boxe']}}
{'_id': '566eec5f662b388eba464207', 'IdSportif': 5, 'Nom': 'COMES', 'Prenom': 'Sylvie', 'Sexe': 'F', 'Age': 22, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Basket ball', 'Volley ball', 'Badmington', 'Ping pong'], 'Arbitrer': 'Ping pong'}}
{'_id': '566eec5f662b388eba464208', 'IdSportif': 6, 'Nom': 'RETALDI', 'Prenom': 'Sophie', 'Sexe': 'F', 'Age': 30, 'IdSportifConseiller': 3, 'Sports': {'Jouer': ['Basket ball', 'Volley ball', 'Hand ball', 'Ping pong'], 'Arbitrer': ['Basket ball', 'Hockey', 'Ping pong'], 'Entrainer': ['Hockey', 'Ping pong', 'Boxe']}}
{'_id': '566eec5f662b388eba46420f', 'IdSportif': 14, 'Nom': 'CAILLIOT', 'P

5) On cherche les sportifs ayant au moins 32 ans, ou de sexe féminin ?

In [9]:
request = sp.find({
    "$or" : [
        {"Age" : 32}, 
        {"Sexe" : "F"}
    ]
})

for element in request :
    print(element)

{'_id': '566eec5f662b388eba464205', 'IdSportif': 3, 'Nom': 'HUE', 'Prenom': 'Pascale', 'Sexe': 'F', 'Age': 25, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Volley ball', 'Ping pong'], 'Arbitrer': ['Volley ball', 'Badmington', 'Ping pong'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Badmington']}}
{'_id': '566eec5f662b388eba464206', 'IdSportif': 4, 'Nom': 'DORLEANS', 'Prenom': 'Jean-michel', 'Sexe': 'M', 'Age': 32, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Volley ball', 'Football'], 'Arbitrer': ['Basket ball', 'Volley ball', 'Ping pong', 'Boxe'], 'Entrainer': ['Basket ball', 'Ping pong', 'Boxe']}}
{'_id': '566eec5f662b388eba464207', 'IdSportif': 5, 'Nom': 'COMES', 'Prenom': 'Sylvie', 'Sexe': 'F', 'Age': 22, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Basket ball', 'Volley ball', 'Badmington', 'Ping pong'], 'Arbitrer': 'Ping pong'}}
{'_id': '566eec5f662b388eba464208', 'IdSportif': 6, 'Nom': 'RETALDI', 'Prenom': 'Sophie', 'Sexe': 'F', 'Age': 30, 'IdSportifConse

6) Trouver le nombre de sportifs de sexe féminin dans la base ?

In [10]:
request = sp.find({
    "Sexe" : "F"
})
print(request.count())

44


7) Quels sont les sportifs (identifiant, nom et prénom) qui ont entre 20 et 30 ans ?

In [11]:
request = sp.find({
    "Age" : {
        "$gte" : 20, 
        "$lt" : 30
    }
},{
    "Nom": 1, 
    "Prenom" : 1
})

for element in request :
    print(element)

{'_id': '566eec5f662b388eba464204', 'Nom': 'KERVADEC', 'Prenom': 'Yann'}
{'_id': '566eec5f662b388eba464205', 'Nom': 'HUE', 'Prenom': 'Pascale'}
{'_id': '566eec5f662b388eba464207', 'Nom': 'COMES', 'Prenom': 'Sylvie'}
{'_id': '566eec5f662b388eba464209', 'Nom': 'GOMEZ', 'Prenom': 'Diego'}
{'_id': '566eec5f662b388eba46420a', 'Nom': 'ANTUNES', 'Prenom': 'Gaêlle'}
{'_id': '566eec5f662b388eba46420b', 'Nom': 'BLANDET', 'Prenom': 'Arnaud'}
{'_id': '566eec5f662b388eba46420c', 'Nom': 'BONNET', 'Prenom': 'Christophe'}
{'_id': '566eec5f662b388eba46420d', 'Nom': 'BORREL', 'Prenom': 'Benoît'}
{'_id': '566eec5f662b388eba46420e', 'Nom': 'BOYON', 'Prenom': 'Julien'}
{'_id': '566eec5f662b388eba46420f', 'Nom': 'CAILLIOT', 'Prenom': 'Stéphanie'}
{'_id': '566eec5f662b388eba464210', 'Nom': 'CHAMPENOIS', 'Prenom': 'Cédric'}
{'_id': '566eec5f662b388eba464211', 'Nom': 'COLAS', 'Prenom': 'Michaël'}
{'_id': '566eec5f662b388eba464212', 'Nom': 'COLOMB', 'Prenom': 'Michaël'}
{'_id': '566eec5f662b388eba464213', 'Nom'

8) Quels sont les sportifs (identifiant et nom) qui pratiquent du hand ball ?

In [12]:
request = sp.find({
    "Sports.Jouer" : "Hand ball"
},{
    "Nom": 1, 
    "Prenom" : 1
})

for element in request :
    print(element)

{'_id': '566eec5f662b388eba464208', 'Nom': 'RETALDI', 'Prenom': 'Sophie'}
{'_id': '566eec5f662b388eba464214', 'Nom': 'TESTEMONT', 'Prenom': 'Laurent'}
{'_id': '566eec5f662b388eba464215', 'Nom': 'TIENER', 'Prenom': 'Thomas'}
{'_id': '566eec5f662b388eba46421a', 'Nom': 'MARTORA', 'Prenom': 'Fabrice'}
{'_id': '566eec5f662b388eba46421b', 'Nom': 'MICHEL', 'Prenom': 'Frédérique'}
{'_id': '566eec5f662b388eba46421c', 'Nom': 'NIELLEZ', 'Prenom': 'Christophe'}
{'_id': '566eec5f662b388eba46421d', 'Nom': 'RABAHI', 'Prenom': 'Rabah'}
{'_id': '566eec5f662b388eba46421e', 'Nom': 'ROUSSEL', 'Prenom': 'Nadège'}
{'_id': '566eec5f662b388eba46421f', 'Nom': 'SCHINK', 'Prenom': 'Nicolas'}
{'_id': '566eec5f662b388eba464220', 'Nom': 'STEMPUT', 'Prenom': 'Mathieu'}
{'_id': '566eec5f662b388eba464221', 'Nom': 'VAN CAUTER', 'Prenom': 'Vincent'}
{'_id': '566eec60662b388eba464222', 'Nom': 'RAMPNOUX', 'Prenom': 'Jean françois'}
{'_id': '566eec60662b388eba464223', 'Nom': 'LHERPINIERE', 'Prenom': 'Olivier'}
{'_id': '566

9) Quels sont les gymnases de “Villetaneuse” ou de “Sarcelles” qui ont une surface de plus de 400 m2 ?

In [13]:
request = gym.find({
    "Ville" : { "$in" : ["VILLETANEUSE", "SARCELLES"]}
})

for element in request :
    print(element)

{'_id': '566eec69662b388eba46429c', 'IdGymnase': 4, 'NomGymnase': 'PAUL ELUARD', 'Adresse': 'Allée J.B. Lulli', 'Ville': 'SARCELLES', 'Surface': 500, 'Seances': [{'IdSportifEntraineur': 149, 'Jour': 'Vendredi', 'Horaire': 10.0, 'Duree': 30, 'Libelle': 'Basket ball'}, {'IdSportifEntraineur': 6, 'Jour': 'mercredi', 'Horaire': 19.0, 'Duree': 60, 'Libelle': 'Hockey'}]}
{'_id': '566eec69662b388eba46429d', 'IdGymnase': 5, 'NomGymnase': 'BRASSENS', 'Adresse': '153 square Loliot', 'Ville': 'SARCELLES', 'Surface': 620, 'Seances': [{'IdSportifEntraineur': 57, 'Jour': 'lundi', 'Horaire': 16.3, 'Duree': 90, 'Libelle': 'Volley ball'}, {'IdSportifEntraineur': 6, 'Jour': 'jeudi', 'Horaire': 19.0, 'Duree': 60, 'Libelle': 'Hockey'}]}
{'_id': '566eec69662b388eba4642a1', 'IdGymnase': 9, 'NomGymnase': 'CAMUS', 'Adresse': '3 esplanade des quatrans', 'Ville': 'SARCELLES', 'Surface': 620, 'Seances': [{'IdSportifEntraineur': 6, 'Jour': 'samedi', 'Horaire': 19.0, 'Duree': 60, 'Libelle': 'Hockey'}]}
{'_id': '56

10) Dans quels gymnases peut-on jouer au hockey le mercredi après 15H ?

In [14]:
request = gym.find({ 
    "$and" : [
        {"Seances.Libelle" : "Hockey"}, 
        {"Seances.Jour" : "mercredi"},
        {"Seances.Horaire" : {"$gte" : 15} }
    ]
})

for element in request :
    print(element)

{'_id': '566eec69662b388eba464299', 'IdGymnase': 1, 'NomGymnase': 'PAUL ELUARD', 'Adresse': '2 rue des pépines', 'Ville': 'STAINS', 'Surface': 200, 'Seances': [{'IdSportifEntraineur': 149, 'Jour': 'Samedi', 'Horaire': 9.0, 'Duree': 60, 'Libelle': 'Basket ball'}, {'IdSportifEntraineur': 1, 'Jour': 'Lundi', 'Horaire': 9.0, 'Duree': 60, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 1, 'Jour': 'Lundi', 'Horaire': 10.0, 'Duree': 60, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 1, 'Jour': 'Lundi', 'Horaire': 11.3, 'Duree': 60, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 1, 'Jour': 'Lundi', 'Horaire': 14.0, 'Duree': 90, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 1, 'Jour': 'lundi', 'Horaire': 17.3, 'Duree': 120, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 1, 'Jour': 'Lundi', 'Horaire': 19.3, 'Duree': 120, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 2, 'Jour': 'Dimanche', 'Horaire': 17.3, 'Duree': 120, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 2, 'Jour': 'Dim

11) Quels sportifs (identifiant et nom) ne pratiquent aucun sport ?

In [15]:
request = sp.find({
    "Sports.Jouer" : { "$exists": False }
},{
    "Nom": 1
})

for element in request :
    print(element)

{'_id': '566eec60662b388eba464252', 'Nom': 'DROULLON'}
{'_id': '566eec60662b388eba464275', 'Nom': 'BELUAU'}
{'_id': '566eec60662b388eba464276', 'Nom': 'FERREIRA'}
{'_id': '566eec60662b388eba464284', 'Nom': 'GALLOIS'}
{'_id': '566eec60662b388eba464292', 'Nom': 'LENEVEU'}
{'_id': '566eec60662b388eba464293', 'Nom': 'DABON'}
{'_id': '566eec60662b388eba464294', 'Nom': 'CLERICE'}
{'_id': '566eec60662b388eba464295', 'Nom': 'COMES'}
{'_id': '566eec60662b388eba464297', 'Nom': 'BELZ'}


120 Quels gymnases n’ont pas de séances le dimanche ?

In [16]:
request = gym.find({
    "Seances.Jour" : {"$ne" : "dimanche"}
})

for element in request :
    print(element)

{'_id': '566eec69662b388eba464299', 'IdGymnase': 1, 'NomGymnase': 'PAUL ELUARD', 'Adresse': '2 rue des pépines', 'Ville': 'STAINS', 'Surface': 200, 'Seances': [{'IdSportifEntraineur': 149, 'Jour': 'Samedi', 'Horaire': 9.0, 'Duree': 60, 'Libelle': 'Basket ball'}, {'IdSportifEntraineur': 1, 'Jour': 'Lundi', 'Horaire': 9.0, 'Duree': 60, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 1, 'Jour': 'Lundi', 'Horaire': 10.0, 'Duree': 60, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 1, 'Jour': 'Lundi', 'Horaire': 11.3, 'Duree': 60, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 1, 'Jour': 'Lundi', 'Horaire': 14.0, 'Duree': 90, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 1, 'Jour': 'lundi', 'Horaire': 17.3, 'Duree': 120, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 1, 'Jour': 'Lundi', 'Horaire': 19.3, 'Duree': 120, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 2, 'Jour': 'Dimanche', 'Horaire': 17.3, 'Duree': 120, 'Libelle': 'Hand ball'}, {'IdSportifEntraineur': 2, 'Jour': 'Dim

13) Quels gymnases ne proposent que des séances de basket ball ou de volley ball ?

In [17]:
request = gym.find({
    'Seances.Libelle' : {
        '$in' : ['Volley ball', 'Basket ball'], 
        '$nin' : ['Hockey', 'Badminton', 'Football', 'Tennis', 'Hand ball']
    }
})

for element in request:
    print(element)

{'_id': '566eec69662b388eba46429a', 'IdGymnase': 2, 'NomGymnase': 'ARAGON', 'Adresse': 'Place du Chartres', 'Ville': 'MONTMORENCY', 'Surface': 450, 'Seances': [{'IdSportifEntraineur': 57, 'Jour': 'dimanche', 'Horaire': 17.0, 'Duree': 60, 'Libelle': 'Volley ball'}]}
{'_id': '566eec69662b388eba46429b', 'IdGymnase': 3, 'NomGymnase': 'SAINT EXUPERY', 'Adresse': '47 bvd des brumes', 'Ville': 'PIERREFITTE', 'Surface': 400, 'Seances': [{'IdSportifEntraineur': 149, 'Jour': 'Mercredi', 'Horaire': 11.0, 'Duree': 30, 'Libelle': 'Basket ball'}, {'IdSportifEntraineur': 57, 'Jour': 'lundi', 'Horaire': 16.3, 'Duree': 90, 'Libelle': 'Volley ball'}, {'IdSportifEntraineur': 60, 'Jour': 'jeudi', 'Horaire': 19.0, 'Duree': 60, 'Libelle': 'Volley ball'}]}
{'_id': '566eec69662b388eba4642a4', 'IdGymnase': 12, 'NomGymnase': 'MOZART', 'Adresse': '6 Allée Rosana', 'Ville': 'MONTMORENCY', 'Surface': 480, 'Seances': [{'IdSportifEntraineur': 57, 'Jour': 'dimanche', 'Horaire': 17.0, 'Duree': 60, 'Libelle': 'Volley b

14)	Quels sont les entraîneurs qui sont aussi joueurs ?


In [18]:
request = sp.find({
    'Sports.Jouer' : { "$exists": True },
    'Sports.Entrainer' : { "$exists": True } 
})

for element in request :
    print(element)

{'_id': '566eec5f662b388eba464203', 'IdSportif': 1, 'Nom': 'BOUTAHAR', 'Prenom': 'Abderahim', 'Sexe': 'm', 'Age': 30, 'Sports': {'Jouer': ['Volley ball', 'Tennis', 'Football'], 'Arbitrer': ['Basket ball', 'Volley ball', 'Hockey'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Hockey', 'Badmington']}}
{'_id': '566eec5f662b388eba464204', 'IdSportif': 2, 'Nom': 'KERVADEC', 'Prenom': 'Yann', 'Sexe': 'M', 'Age': 28, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Basket ball', 'Volley ball', 'Ping pong', 'Football'], 'Arbitrer': ['Hockey', 'Football'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Tennis', 'Hockey', 'Badmington', 'Ping pong', 'Boxe']}}
{'_id': '566eec5f662b388eba464205', 'IdSportif': 3, 'Nom': 'HUE', 'Prenom': 'Pascale', 'Sexe': 'F', 'Age': 25, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Volley ball', 'Ping pong'], 'Arbitrer': ['Volley ball', 'Badmington', 'Ping pong'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Badmington']}}
{'_i

15)	Quels sont les sportifs qui sont des conseillers ?


In [19]:
request = sp.find({
    'Sports.Jouer' : { "$exists": True },
    'IdSportif' : { '$in' : sp.find({ 
        'IdSportifConseiller' : { "$exists": True }
    }).distinct('IdSportifConseiller')}
})

for element in request :
    print(element)

{'_id': '566eec5f662b388eba464203', 'IdSportif': 1, 'Nom': 'BOUTAHAR', 'Prenom': 'Abderahim', 'Sexe': 'm', 'Age': 30, 'Sports': {'Jouer': ['Volley ball', 'Tennis', 'Football'], 'Arbitrer': ['Basket ball', 'Volley ball', 'Hockey'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Hockey', 'Badmington']}}
{'_id': '566eec5f662b388eba464204', 'IdSportif': 2, 'Nom': 'KERVADEC', 'Prenom': 'Yann', 'Sexe': 'M', 'Age': 28, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Basket ball', 'Volley ball', 'Ping pong', 'Football'], 'Arbitrer': ['Hockey', 'Football'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Tennis', 'Hockey', 'Badmington', 'Ping pong', 'Boxe']}}
{'_id': '566eec5f662b388eba464205', 'IdSportif': 3, 'Nom': 'HUE', 'Prenom': 'Pascale', 'Sexe': 'F', 'Age': 25, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Volley ball', 'Ping pong'], 'Arbitrer': ['Volley ball', 'Badmington', 'Ping pong'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Badmington']}}
{'_i

16)	Pour le sportif “Kervadec” quel est le nom de son conseiller ?

In [39]:
request = sp.find({
    'IdSportif' : sp.find({
        'Nom' : "KERVADEC"
    })[0]['IdSportifConseiller']
})

for element in request :
    print(element)

{'_id': '566eec5f662b388eba464203', 'IdSportif': 1, 'Nom': 'BOUTAHAR', 'Prenom': 'Abderahim', 'Sexe': 'm', 'Age': 30, 'Sports': {'Jouer': ['Volley ball', 'Tennis', 'Football'], 'Arbitrer': ['Basket ball', 'Volley ball', 'Hockey'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Hockey', 'Badmington']}}


17)	Quels entraîneurs entraînent du hand ball et du basket ball ?

In [40]:
request = sp.find({
    'Sports.Entrainer' : { '$in' : ['Hand ball', 'Basket ball']}
})

for element in request :
    print(element)

{'_id': '566eec5f662b388eba464203', 'IdSportif': 1, 'Nom': 'BOUTAHAR', 'Prenom': 'Abderahim', 'Sexe': 'm', 'Age': 30, 'Sports': {'Jouer': ['Volley ball', 'Tennis', 'Football'], 'Arbitrer': ['Basket ball', 'Volley ball', 'Hockey'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Hockey', 'Badmington']}}
{'_id': '566eec5f662b388eba464204', 'IdSportif': 2, 'Nom': 'KERVADEC', 'Prenom': 'Yann', 'Sexe': 'M', 'Age': 28, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Basket ball', 'Volley ball', 'Ping pong', 'Football'], 'Arbitrer': ['Hockey', 'Football'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Tennis', 'Hockey', 'Badmington', 'Ping pong', 'Boxe']}}
{'_id': '566eec5f662b388eba464205', 'IdSportif': 3, 'Nom': 'HUE', 'Prenom': 'Pascale', 'Sexe': 'F', 'Age': 25, 'IdSportifConseiller': 1, 'Sports': {'Jouer': ['Volley ball', 'Ping pong'], 'Arbitrer': ['Volley ball', 'Badmington', 'Ping pong'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Badmington']}}
{'_i

18)	Quels sportifs n’ont pas de conseillers ?

In [43]:
request = sp.find({
    'IdSportifConseiller' : { '$exists' : False }
})

for element in request :
    print(element)

{'_id': '566eec5f662b388eba464203', 'IdSportif': 1, 'Nom': 'BOUTAHAR', 'Prenom': 'Abderahim', 'Sexe': 'm', 'Age': 30, 'Sports': {'Jouer': ['Volley ball', 'Tennis', 'Football'], 'Arbitrer': ['Basket ball', 'Volley ball', 'Hockey'], 'Entrainer': ['Basket ball', 'Volley ball', 'Hand ball', 'Hockey', 'Badmington']}}
{'_id': '566eec60662b388eba46422e', 'IdSportif': 45, 'Nom': 'CHAVANT', 'Prenom': 'Christophe', 'Sexe': 'M', 'Age': 25, 'Sports': {'Jouer': ['Tennis', 'Football']}}
{'_id': '566eec60662b388eba46423c', 'IdSportif': 59, 'Nom': 'CLERICE', 'Prenom': 'Stéphanie', 'Sexe': 'F', 'Age': 23, 'Sports': {'Jouer': ['Basket ball', 'Badmington', 'Ping pong'], 'Arbitrer': 'Tennis', 'Entrainer': ['Volley ball', 'Tennis']}}
{'_id': '566eec60662b388eba464242', 'IdSportif': 65, 'Nom': 'CHESNIER', 'Prenom': 'Marc', 'Sexe': 'M', 'Age': 30, 'Sports': {'Jouer': 'Football'}}
{'_id': '566eec60662b388eba46424f', 'IdSportif': 78, 'Nom': 'CHAUVIN', 'Prenom': 'Julien', 'Sexe': 'M', 'Age': 30, 'Sports': {'Jou

19) Votre propre Qry : la plus intéressante et/ou la plus complexe...

In [44]:
request = sp.find()

for element in request :
    print(element)

portifConseiller': 4, 'Sports': {'Jouer': ['Tennis', 'Football']}}
{'_id': '566eec60662b388eba464230', 'IdSportif': 47, 'Nom': 'GROEN', 'Prenom': 'Céline', 'Sexe': 'F', 'Age': 25, 'IdSportifConseiller': 2, 'Sports': {'Jouer': 'Tennis'}}
{'_id': '566eec60662b388eba464231', 'IdSportif': 48, 'Nom': 'HEDDI', 'Prenom': 'Zohra', 'Sexe': 'F', 'Age': 23, 'IdSportifConseiller': 2, 'Sports': {'Jouer': ['Basket ball', 'Badmington', 'Ping pong'], 'Entrainer': 'Badmington'}}
{'_id': '566eec60662b388eba464232', 'IdSportif': 49, 'Nom': 'JOUVE', 'Prenom': 'Sandra', 'Sexe': 'F', 'Age': 24, 'IdSportifConseiller': 5, 'Sports': {'Jouer': ['Basket ball', 'Ping pong']}}
{'_id': '566eec60662b388eba464233', 'IdSportif': 50, 'Nom': 'KALOMBO', 'Prenom': 'Yannick', 'Sexe': 'M', 'Age': 22, 'IdSportifConseiller': 2, 'Sports': {'Jouer': ['Basket ball', 'Badmington', 'Ping pong', 'Football'], 'Entrainer': 'Badmington'}}
{'_id': '566eec60662b388eba464234', 'IdSportif': 51, 'Nom': 'LOPEZ', 'Prenom': 'Thibaud', 'Sexe':

20) 1 point offert

In [ ]:
print("Merci")